hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -1174,7 +1178,7 @@ public HStore call() throws IOException {
           LOG.info("Setting FlushNonSloppyStoresFirstPolicy for the region=" + this);
         }
       } catch (InterruptedException e) {
-        throw (InterruptedIOException)new InterruptedIOException().initCause(e);
+        throwOnInterrupt(e);
```

Comment:
Better make it return an exception and make here 'throw throwOnInterrupt(e)' so the compiler will know that this is an exit point.

Reply:
Ok